xarray/plot/dataarray_plot.py

```diff
@@ -524,7 +524,7 @@ def line(
         assert hueplt is not None
         ax.legend(handles=primitive, labels=list(hueplt.to_numpy()), title=hue_label)
 
-    if np.issubdtype(xplt.dtype, np.datetime64):
+    if isinstance(xplt.dtype, np.dtype) and np.issubdtype(xplt.dtype, np.datetime64):  # type: ignore[redundant-expr]
```

Comment:
Why was this necessary?

Reply:
`dtype` from ` DataArray` is typed as being `numpy` but in reality, `NamedArray`'s backing data can have pandas extension array dtype.  How this leaks out of the typing is a different story.  I think that would be worth investigating

Reply:
And using `np.issubdtype` does not work on extension arrays?

Reply:
Right, it complains about LHS not being a numpy dtype